lightning/src/util/sweep.rs

```diff
@@ -0,0 +1,688 @@
+	/// # use lightning::chain::{BestBlock, Filter, WatchedOutput};
+	/// # use lightning::util::sweep::OutputSweeper;
+	/// # use lightning::util::logger::{Logger, Record};
+	/// # use lightning::io;
+	/// # struct MyStore {}
+	/// # impl lightning::util::persist::KVStore for MyStore {
+	/// #     fn read(&self, primary_namespace: &str, secondary_namespace: &str, key: &str) -> io::Result<Vec<u8>> { Ok(Vec::new()) }
+	/// #     fn write(&self, primary_namespace: &str, secondary_namespace: &str, key: &str, buf: &[u8]) -> io::Result<()> { Ok(()) }
+	/// #     fn remove(&self, primary_namespace: &str, secondary_namespace: &str, key: &str, lazy: bool) -> io::Result<()> { Ok(()) }
+	/// #     fn list(&self, primary_namespace: &str, secondary_namespace: &str) -> io::Result<Vec<String>> { Ok(Vec::new()) }
+	/// # }
+	/// # struct MyWallet {}
+	/// # impl MyWallet {
+	/// #     fn get_new_address(&self) -> ScriptBuf { ScriptBuf::new() }
+	/// # }
+	/// # struct MyFeeEstimator {}
+	/// # impl FeeEstimator for MyFeeEstimator {
+	/// #     fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 { 0 }
+	/// # }
+	/// # struct MyBroadcaster {}
+	/// # impl BroadcasterInterface for MyBroadcaster {
+	/// #     fn broadcast_transactions(&self, txs: &[&Transaction]) {}
+	/// # }
+	/// # struct MyFilter {}
+	/// # impl Filter for MyFilter {
+	/// #     fn register_tx(&self, txid: &Txid, script_pubkey: &Script) {}
+	/// #     fn register_output(&self, output: WatchedOutput) {}
+	/// # }
+	/// # struct MyLogger {}
+	/// # impl Logger for MyLogger {
+	/// #     fn log(&self, record: Record) {}
+	/// # }
+	/// # let my_wallet = Arc::new(MyWallet{});
+	/// # let my_fee_estimator = Arc::new(MyFeeEstimator{});
+	/// # let my_broadcaster = Arc::new(MyBroadcaster {});
+	/// # let my_keys_manager = Arc::new(KeysManager::new(&[42u8; 32], 0, 0));
+	/// # let my_store = Arc::new(MyStore {});
+	/// # let my_logger = Arc::new(MyLogger {});
+	/// # let my_best_block = BestBlock::from_network(Network::Regtest);
+	/// # let my_chain_data_source = Arc::new(MyFilter {});
+	/// let spend_wallet = Arc::clone(&my_wallet);
+	/// let spend_keys_manager = Arc::clone(&my_keys_manager);
+	/// let spend_fee_estimator = Arc::clone(&my_fee_estimator);
+	/// let spend_outputs_callback = move |output_descriptors: &[&SpendableOutputDescriptor]| {
+	/// 	let change_destination_script = spend_wallet.get_new_address();
+	/// 	let fee_rate = spend_fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::NonAnchorChannelFee);
+	/// 	spend_keys_manager.spend_spendable_outputs(
+	/// 		output_descriptors,
+	/// 		Vec::new(),
+	/// 		change_destination_script,
+	/// 		fee_rate,
+	/// 		None,
+	/// 		&Secp256k1::new(),
+	/// 	)
+	/// };
+	///
+	/// let sweeper = OutputSweeper::new(my_broadcaster, my_keys_manager, my_store, my_best_block, Some(my_chain_data_source), my_logger, None, spend_outputs_callback);
```

Comment:
Could be more concise if you drop all the `my_` prefixes, which don't really add anything.